Update release docs (#15555)
apache/beam: website/www/site/content/en/contribute/release-guide.md

If there is a bug found in the RC creation process/tools, those issues should be considered high priority and fixed in 7 days.

> ### Process for cherry-picks

Reviewer: Nit: I don't think we need a new section for this, we can just add the first bullet point to the existing "Review cherry-picks" section.

Author: Sounds Good!

> * Feature owner should first merge the changes in master and then create the cherry-pick from the merge commit against release branch. The cr should start have `[cherrypick]` in the title and release manager as the reviewer.

Reviewer: typo?
```suggestion
* Feature owner should first merge the changes in master and then create the cherry-pick from the merge commit against release branch. The PR should have `[cherrypick]` in the title and release manager as the reviewer.
```

     * Copy python doc into beam-site
     * Copy java doc into beam-site

* **Mitigate GPG Out Of Memory Error**

>   * gpg-agent might through Out Of Memory error. You can kill the gpg-agent to mitigate this issue.

Reviewer: ```suggestion
  * gpg-agent might throw Out Of Memory error. You can kill the gpg-agent to mitigate this issue.
```

Reviewer: Is there a jira for this? killing the agent is a workaround, but we should also plan to fix this permanently.

Reviewer 2: I think I saw this as well, but I was also having issues with the gpg passphrase prompt appearing while gradle was running (in ssh, garbling the output of both gradle and gpg and the passphrase prompt was not reading all the characters).

My solution was to use `gradlew  --no-parallel` which also solved the OOM.

Reviewer 2: https://github.com/apache/beam/pull/15544/files#diff-b47163558f49ed59a8a91deaa893c4bae776bfc9bd6249a64814ed7acfcc2621R181

         Artifact names should follow [the existing format](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22org.apache.beam%22) in which artifact name mirrors directory structure, e.g., `beam-sdks-java-io-kafka`.
         Carefully review any new artifacts.

> __NOTE__: When uploading to Nexus using gradle, ip address is used as a seesion id. Incorrect partial artifact upload will result if the public ip of the dev machine changes during upload.

Reviewer: ```suggestion
__NOTE__: When uploading to Nexus using gradle, ip address is used as a session id. Incorrect partial artifact upload will result if the public ip of the dev machine changes during upload.
```

Reviewer: Please link the jira for this issue here if there is one.

> __NOTE__: If you notice multiple Nexus staging repositories getting created, try reducing the parallelism of gradle upload task.

Reviewer 2: I plan on disabling parallelism for the Nexus upload stage in my update to the script.

Author: I see that then we don't need this section.
